bundles/http/Citrix/Covid-19-Self-Certify/script.js

```diff
@@ -0,0 +1,219 @@
+                "confirmation": actionParameters.status,
+                "createdon": {
+                    "start": moment().format('YYYY-MM-DD HH:mm:ss')
+                },
+                "country": [{ "value": actionParameters.country }],
+                "city": [{ "value": actionParameters.city }]
+            }
+        })
+    })
+    if (!response.ok) {
+        throw new Error(await response.text())
+    }
+    await getItems(client, dataStore, integrationParameters, true)
+}
+
+async function getRegionData(client, dataStore, integrationParameters) {
+    const limit = 100
+    let offset = 0
+    let total = 0
+    do {
+        const countryDataResponse = await client.fetch(`item/app/${integrationParameters.countryDataAppId}/filter`, {
+            method: "POST",
+            body: JSON.stringify({
+                limit: limit,
+                offset: offset
+            })
+        })
+        const {responseBody,errorMessage,errorStatus} = await validateResponse(countryDataResponse)
+        if (errorStatus) {
+            console.log(JSON.stringify(responseBody),errorMessage)
+            throw new Error(errorMessage)
+        }
+        total = responseBody.total
+        const items = responseBody.items ?? []
+        items.forEach(item => {
+            const fields = item.fields ?? []
+            const region = getTextValue(item.fields, 'text', 'Region') ?? null
+            const countries = String(getTextValue(item.fields, 'text', 'Countries')).split(',')
+            countries?.forEach(country => {
+                dataStore.save("cities", {
+                    "uniqueId": uuid.v4(),
+                    "countryName": country,
+                    "regionName": region
+                })
+            })
+
+            dataStore.save('countryConditions', {
+                "uniqueId": uuid.v4(),
+                "region": region,
+                "guidelines": escapeString(getTextValue(item.fields,'text','Guidelines')) ?? null,
+                "image":getImage(item.fields,'Image') ?? null
+            })
+
+        })
+        offset += limit
+    } while (offset <= total)
+}
+
+function getTextValue(dataItems, type, label = null) {
+    const item = dataItems.find(item => (label != null && item.label == label && item.type === type) || (label == null && item.type == type))
+    return item?.values[0]?.value ?? null
+}
+
+function getCategoryValue(dataItems) {
+    const confirmation = dataItems.find(item => item.type === 'category')
+    return confirmation?.values[0]?.value?.text ?? null
+}
+
+function escapeString(conditions) {
+    return conditions?.replaceAll('<br/>', '\r\n').replaceAll('</p>', '\r\n').replaceAll('<p>', '') ?? null
```

Comment:
better function name: this algorithm is some html to plain text but definitely not string escaping

Reply:
Function name has been changed to `convertHtmlTagsToStringEscapeSequence`